### PR TITLE
Migrate Input and ControlsGroup to TypeScript

### PR DESCRIPTION
**Pull request by loonywizard**
https://kupibiletru.atlassian.net/browse/SITE-2957

### CODE REVIEW

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-8c78b7e4`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-8d6d2a3d`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-691e5d40`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-fc19d186`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-6858115b`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-4a6b761c`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.0-SITE-2957-migrate-input-to-ts-a4090e19`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.1-SITE-2957-migrate-input-to-ts-3f98f9fe`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-50b067e7`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-2f13b7f1`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-c1d548ef`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-ccbb7b79`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-46e844cf`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-6ff048cc`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-cfd3f0e8`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-51b52a64`

File: src/components/Input/index.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  const handleFocus = useCallback((
+    onFocusArg: ((event: React.FocusEvent<T>) => void) | null, event: React.FocusEvent<T>,
+  ): void => {
+    if (onFocus) onFocus(event)
+    if (onFocusArg) onFocusArg(event)
+
+    setIsActive(true)
+  }, [])
+
+  const handleBlur = useCallback((
```
[the remark this answers is not in the document]
Comment (loonywizard) [reply src/components/Input/index.tsx]: согласен, убрал

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-3bbfb28b`

**Review by privetkometa — APPROVED** [review]

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-497d001c`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-1c597093`

**Comment by Kupibilet-System** [comment]
New version of package: `12.1.2-SITE-2957-migrate-input-to-ts-bb410f9e`

**Comment by Kupibilet-System** [comment]
New version of package: `12.2.0-SITE-2957-migrate-input-to-ts-036ee2a6`

**Pull request merged**